deploy/kubernetes/webhook-example/README.md

```diff
@@ -0,0 +1,67 @@
+# How to deploy the webhook
+
+The webhook server is provided as an image which can be built from this repository. It can be deployed anywhere, as long as the api server is able to reach it over HTTPS. It is recommended to deploy the webhook server in the cluster as snapshotting is latency sensitive. A `ValidatingWebhookConfiguration` object is needed to configure the api server to contact the webhook server. Please see the [documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) for more details. The webhook server code is adapted from the [webhook server](https://github.com/kubernetes/kubernetes/tree/v1.18.6/test/images/agnhost/webhook) used in the kubernetes/kubernetes end to end testing code.
+
+## Example in-cluster deployment using Kubernetes Secrets
+
+Please note this is not considered to be a production ready method to deploy the certificates and is only provided for demo purposes. This is only one of many ways to deploy the certificates, it is your responsibility to ensure the security of your cluster. TLS certificates and private keys should be handled with care and you may not want to keep them in plain Kubernetes secrets. These commands should be run from the top level directory.
+
+This method was heavily adapted from [banzai cloud](https://banzaicloud.com/blog/k8s-admission-webhooks/).
+
+### Method
+
+Run the `create-cert.sh` script. Note using the default namespace will allow anyone with access to that namespace to read your secret. It is recommended to change the namespace in all the file and the command.
+
+This script will create a TLS certificate signed by the [cluster](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/). It will place the public and private key into a secret on the cluster.
+
+```bash
+./deploy/kubernetes/webhook-example/create-cert.sh --service snapshot-validation-service --namespace default --secret snapshot-validation-secret
```

Comment:
I'd make the cmd like: --namespace <secret-namespace> --secret <secret-name> and change the security implication in line 13 to not to refer a specific namespace "default"? However this would require the admission-configuration-template to also have the "default" namespace substituted by "patch-ca-bundle.sh"